exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverExportUtils.java

```diff
@@ -374,54 +377,77 @@ static MonitoredResource getDefaultResource() {
-      case GCP_GKE_CONTAINER:
-        GcpGkeContainerMonitoredResource gcpGkeContainerMonitoredResource =
-            (GcpGkeContainerMonitoredResource) autoDetectedResource;
-        builder.putLabels(PROJECT_ID_LABEL_KEY, gcpGkeContainerMonitoredResource.getAccount());
-        builder.putLabels("cluster_name", gcpGkeContainerMonitoredResource.getClusterName());
-        builder.putLabels("container_name", gcpGkeContainerMonitoredResource.getContainerName());
-        builder.putLabels("namespace_name", gcpGkeContainerMonitoredResource.getNamespaceId());
-        builder.putLabels("pod_name", gcpGkeContainerMonitoredResource.getPodId());
-        builder.putLabels("location", gcpGkeContainerMonitoredResource.getZone());
-        return;
-      case AWS_EC2_INSTANCE:
-        AwsEc2InstanceMonitoredResource awsEc2InstanceMonitoredResource =
-            (AwsEc2InstanceMonitoredResource) autoDetectedResource;
-        builder.putLabels("aws_account", awsEc2InstanceMonitoredResource.getAccount());
-        builder.putLabels("instance_id", awsEc2InstanceMonitoredResource.getInstanceId());
-        builder.putLabels("region", "aws:" + awsEc2InstanceMonitoredResource.getRegion());
+    Map<String, String> mappings = null;
+    switch (type) {
+      case ResourceKeyConstants.GCP_GCE_INSTANCE_TYPE:
+        builder.setType(GCP_GCE_INSTANCE);
+        mappings = GCP_RESOURCE_MAPPING;
+        break;
+      case ResourceKeyConstants.GCP_GKE_INSTANCE_TYPE:
+        builder.setType(GCP_GKE_CONTAINER);
+        mappings = GKE_RESOURCE_MAPPING;
+        break;
+      case ResourceKeyConstants.AWS_EC2_INSTANCE_TYPE:
+        builder.setType(AWS_EC2_INSTANCE);
+        mappings = AWS_RESOURCE_MAPPING;
+        break;
+      default:
+        builder.setType(GLOBAL);
         return;
     }
+
+    if (mappings != null) {
+      Map<String, String> resLabels = autoDetectedResource.getLabels();
+      for (Map.Entry<String, String> entry : mappings.entrySet()) {
+        if (resLabels.containsKey(entry.getValue())) {
+          builder.putLabels(entry.getKey(), resLabels.get(entry.getValue()));
+        }
+      }
+      return;
+    }
+
     throw new IllegalArgumentException("Unknown subclass of MonitoredResource.");
   }
 
+  private static Map<String, String> getGcpResourceLabelsMappings() {
+    Map<String, String> resourceLabels = new LinkedHashMap<String, String>();
+    resourceLabels.put(PROJECT_ID_LABEL_KEY, ResourceKeyConstants.GCP_ACCOUNT_ID_KEY);
+    resourceLabels.put("instance_id", ResourceKeyConstants.GCP_INSTANCE_ID_KEY);
+    resourceLabels.put("zone", ResourceKeyConstants.GCP_ZONE_KEY);
+    return resourceLabels;
+  }
+
+  private static Map<String, String> getGkeResourceLabelsMappings() {
+    Map<String, String> resourceLabels = new LinkedHashMap<String, String>();
+    resourceLabels.put(PROJECT_ID_LABEL_KEY, ResourceKeyConstants.GCP_ACCOUNT_ID_KEY);
+    resourceLabels.put("location", ResourceKeyConstants.GCP_GKE_ZONE_KEY);
```

Comment:
"project_id" and "location" have been reason for a lot of controversy.
For some resources, they are part of the identity, e.g. GCP or AWS instance. But for the "k8s_container" resource, they primarily indicate where the signal data should be stored.

@bogdandrutu ultimately suggested that for resources we should not derive those values from resource detection but rather from static configuration of the Stackdriver exporter.  
For most cases the outcome is probably equivalent, but I think it makes a lot of sense regardless once we look at on-prem clusters.

I have an attempt at this [here](https://github.com/census-ecosystem/opencensus-go-exporter-stackdriver/compare/resource...fabxc:resource#diff-47aceb412c41a069374790bf4dc6de1dR283). To keep implementation simple, the exporter's configuration is written das resource labels as well, so the mappings can still be purely based on labels.
This hasn't been reviewed or agreed on yet – but worth considering here.